services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py

```diff
@@ -155,6 +163,21 @@ async def _get_data_from_port(port: Port) -> Tuple[Port, ItemConcreteValue]:
 
 
 async def download_inputs(inputs_path: Path, port_keys: List[str]) -> ByteSize:
+    return await _download_target_port(
+        port_type_name="inputs", target_path=inputs_path, port_keys=port_keys
+    )
+
+
+async def download_outputs(outputs_path: Path, port_keys: List[str]) -> ByteSize:
+    return await _download_target_port(
+        port_type_name="outputs", target_path=outputs_path, port_keys=port_keys
+    )
+
+
+async def _download_target_port(
+    port_type_name: str, target_path: Path, port_keys: List[str]
+) -> ByteSize:
+    assert port_type_name in {"inputs", "outputs"}
```

Comment:
Append ``nosec`` at the end before your "friend" codeclimate tells you to do it

TIP: I would also add an annotation so the static analysis tool warns you if used explicitly, otherwise the assert warns you at dev/test runtime:

```python
 async def _download_target_port(port_type_name: Literal["inputs", "outputs"],  ... ) :
      assert port_type_name in {"inputs", "outputs"} # nosec
      ...
```

Reply:
even better, issue no longer present after accepting SAN's refactor proposal.